### PR TITLE
Symintify repeat_interleave

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109133

### CODE REVIEW

File: aten/src/ATen/native/Repeat.h [hunk 1]
```diff
@@ -17,7 +17,7 @@ template <
     void compute(index_t*, int64_t*, index_t*, int64_t, int64_t)>
 static inline Tensor repeat_interleave_common(
     const Tensor& repeats,
-    c10::optional<int64_t> output_size) {
+    c10::optional<SymInt> output_size) {
```
Comment (voznesenskym) [inline aten/src/ATen/native/Repeat.h:20]: This does not suffice - the `int64_t` below needs to be a c10::SymInt now, empty needs to be empty_symint, the `compute` template needs to be rewritten for symints, etc etc - you need to symintify the whole impl to match the body.

**Review by ezyang — APPROVED** [review]
I was expecting a test to stop xfailing though

File: aten/src/ATen/functorch/BatchRulesDecompositions.cpp [hunk 2]
```diff
@@ -207,8 +207,8 @@ TORCH_LIBRARY_IMPL(aten, FuncTorchBatchedDecomposition, m) {
   OP_DECOMPOSE(positive);
   OP_DECOMPOSE(qr);
   OP_DECOMPOSE(ravel);
-  m.impl("repeat_interleave.self_int", native::repeat_interleave_symint);
-  OP_DECOMPOSE2(repeat_interleave, self_Tensor);
+  m.impl("repeat_interleave.self_int", static_cast<decltype(&ATEN_FN2(repeat_interleave, self_int))>(native::repeat_interleave_symint));
+  m.impl("repeat_interleave.self_Tensor", static_cast<decltype(&ATEN_FN2(repeat_interleave, self_Tensor))>(native::repeat_interleave_symint));
```
Comment (voznesenskym) [inline aten/src/ATen/functorch/BatchRulesDecompositions.cpp:211]: oh nice

**Review by voznesenskym — APPROVED** [review]

**Review by bdhirsh — APPROVED** [review]

**Pull request closed without merging**